mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -45,37 +49,73 @@ pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx
+        .get_tokens_to_disable(&ticker)
+        .await
+        .into_iter()
+        .chain(std::iter::once(ticker.clone()))
+        .collect::<Vec<_>>();
```

Comment:
I believe this is simpler than using iterators.
``` rust
    // If a platform coin is to be disabled, we get all the enabled tokens for this platform coin first.
    let mut coins_to_disable = coins_ctx
        .get_tokens_to_disable(&ticker)
        .await;
    // We then add the platform coin to the end of the list of the coins to be disabled.
    coins_to_disable.push(ticker.clone());
    drop_mutability!(coins_to_disable);
```

Reply:
Good note!

Reply:
thanks